modules/tinymce/src/plugins/table/main/ts/ui/UiUtils.ts

```diff
@@ -69,6 +69,14 @@ const buildListItems = (items: UserListItem[]): Dialog.ListBoxItemSpec[] =>
     }
   });
 
+export const buildClassList = (classList: UserListItem[]): Dialog.ListBoxItemSpec[] => {
+  if (!classList.length) {
+    return [];
+  }
+
+  return buildListItems([{ text: 'Select...', value: 'mce-no-match' }, ...classList ]);
+};
+
```

Comment:
```suggestion
export const buildClassList = (classList: UserListItem[]): Dialog.ListBoxItemSpec[] => 
  Optionals.someIf(classList.length !== 0, buildListItems([{ text: 'Select...', value: 'mce-no-match' }, ...classList ]));

```
`buildClassList` could return an optional, and then where it's used you could use:
```
const classes = UiUtils.buildClassList(Options.getTableClassList(editor));

  if (classes.isSome()) {
    if (data.class) {
      data.class = data.class.replace(/\s*mce\-item\-table\s*/g, '');
    }
  }
```
and 
```
const getClassList = (editor: Editor): Optional<Dialog.ListBoxSpec> =>
  UiUtils.buildClassList(Options.getRowClassList(editor)).map((classes) => ({
      name: 'class',
      type: 'listbox',
      label: 'Class',
      items: classes
    }));
```

Reply:
Good point. I refactored `buildClassList` to return an `Optional` instead but didn't use `someIf`, since it would call `buildListItems` regardless of `classList.length` is. Maybe if there was some lazy version of `someIf` or something similar, but kind of seems a bit over engineered for what should just be one if statement to early return.